app/plugin_requests.go

```diff
@@ -89,6 +89,11 @@ func (a *App) ServePluginPublicRequest(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	pluginID := vars["plugin_id"]
 
+	if a.GetPluginsEnvironment() == nil {
+		http.NotFound(w, r)
+		return
+	}
```

Comment:
I guess my question would be if we have any idea on why the plugin environment would be nil here. Is it a race condition of some sort? (e.g. not yet initialized).

Reply:
I should have written that down on the description, my bad. The `GetPluginsEnvironment` check the config to see if `PluginSettings` are enabled. If not it will return nil. So we have a "race" between an active request that hits the server for the plugin api and someone disabling the plugin settings. https://github.com/mattermost/mattermost-server/blob/master/app/plugin.go#L45-L47

I think that Is not worth synchronizing those conditions because they have different critical sections.

Reply:
Awesome, thank you for the added context :+1: . Would it make sense to leave a small comment in the code? I am sure it would help future readers.